terminus-ui/drawer/src/drawer.component.ts

```diff
@@ -0,0 +1,343 @@
+        }
+    }`,
+  },
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+  exportAs: 'tsDrawer',
+})
+export class TsDrawerComponent implements AfterContentChecked, OnDestroy {
+
+  public animationState: 'open-instant' | 'open' | 'void' = 'void';
+
+  // Emits whenever the drawer has started animating.
+  public animationStarted = new Subject<AnimationEvent>();
+
+  // Emits whenever the drawer is done animating.
+  public animationEnd = new Subject<AnimationEvent>();
+
+  // Emits when the component is destroyed.
+  private readonly destroyed = new Subject<void>();
+
+  // Whether the drawer is initialized. Used for disabling the initial animation.
+  private enableAnimations = false;
+
+  /**
+   * An observable that emits when the drawer mode changes. This is used by the drawer container to
+   * to know when the mode changes so it can adapt the margins on the content.
+   */
+  public readonly modeChanged = new Subject();
+
+  public get width(): number {
+    return this.elementRef.nativeElement ? (this.elementRef.nativeElement.offsetWidth || 0) : 0;
+  }
+
+  // aria role label, default to nothing
+  @Input()
+  public role = '';
+
+  // Collapsed drawer width
+  @Input()
+  public set collapsedSize(value: string) {
+    this._collapsedSize = value;
+  }
+  public get collapsedSize() {
+    return this._collapsedSize;
+  }
+  public _collapsedSize = '3.75rem';
+
+  // Expanded drawer width
+  @Input()
+  public set expandedSize(value: string) {
+    this._expandedSize = value;
+  }
+  public get expandedSize() {
+    return this._expandedSize;
+  }
+  public _expandedSize = '12.75rem';
+
+  // The side that the drawer is attached to.
+  @Input()
+  public set position(value: TsDrawerPosition) {
+    // Make sure we have a valid value.
+    value = value === 'end' ? 'end' : 'start';
+    if (value !== this._position) {
+      this._position = value;
+      this.onPositionChanged.emit();
+    }
+  }
+  public get position(): TsDrawerPosition {
+    return this._position;
+  }
+  private _position: TsDrawerPosition = 'start';
+
+  // Mode of the drawer, overlay or push
+  @Input()
+  public set mode(value: TsDrawerModes) {
+    this._mode = value;
+    this.modeChanged.next();
+  }
+  public get mode(): TsDrawerModes {
+    return this._mode;
+  }
+  private _mode: TsDrawerModes = 'overlay';
+
+  /**
+   * Whether the drawer is opened. We overload this because we trigger an event when it
+   * starts or end.
+   */
+  @Input()
+  public set isExpanded(value: boolean) {
+    this.toggle(coerceBooleanProperty(value));
+  }
+  public get isExpanded(): boolean {
+    return this._isExpanded;
+  }
+  private _isExpanded = false;
+
+  // Event emitted when the drawer open state is changed.
+  @Output() public readonly expandedChange: EventEmitter<boolean> =
+    // Note this has to be async in order to avoid some issues with two-bindings.
+    new EventEmitter<boolean>(/* isAsync */true);
```

Comment:
Does this mean that the `isAsync` comment actually has an impact on the compilation?

Also, can use the simpler method of defining emitters where the property doesn't have a defined type: `expandedChange = new EventEmitter<boolean>(true);`

Reply:
Ah. Yeah, if this is a needed note about the Emitter, I'd move it to the Emitter comment above. IMO the inline component is odd just cause we don't do that anywhere.